sysdig/getmetrics_sysdig.py

```diff
@@ -0,0 +1,348 @@
+    formated_data = []
+
+    for data_dict in res['data']:
+
+        instance = data_dict['d'][0] + '_' + data_dict['d'][1]
+
+        metric_data_dict={}
+
+        if sub_metric_list== ['container.id','host.hostName']:
+            iterator =0
+        else:
+            iterator =2
+
+        for metric in sub_metric_list:
+            # if(metric not in ['container.id','host.hostName']):
+                metric_data_dict[metric + '[' + instance + ']'] = str(data_dict['d'][iterator])
+                iterator += 1
+
+        metric_data_dict['timestamp']=str(data_dict['t']*1000)
+        formated_data.append(metric_data_dict)
+
+    return formated_data
+
+
+def send_data(chunk_metric_data):
+    send_data_time = time.time()
+    # prepare data for metric streaming agent
+    to_send_data_dict = dict()
+    to_send_data_dict["metricData"] = json.dumps(chunk_metric_data)
+    to_send_data_dict["licenseKey"] = agent_config_vars['licenseKey']
+    to_send_data_dict["projectName"] = agent_config_vars['projectName']
+    to_send_data_dict["userName"] = agent_config_vars['userName']
+    to_send_data_dict["instanceName"] = socket.gethostname().partition(".")[0]
+    to_send_data_dict["samplingInterval"] = agent_config_vars['samplingInterval']
+    to_send_data_dict["agentType"] = "custom"
+
+    to_send_data_json = json.dumps(to_send_data_dict)
+    print(json.dumps(to_send_data_dict))
+    logger.debug("TotalData: " + str(len(bytearray(to_send_data_json))))
+
+    # send the data
+    post_url = parameters['serverUrl'] + "/customprojectrawdata"
+    for _ in xrange(ATTEMPTS):
+        try:
+            if len(if_proxies) == 0:
+                response = requests.post(post_url, data=json.loads(to_send_data_json))
+            else:
+                response = requests.post(post_url, data=json.loads(to_send_data_json), proxies=if_proxies)
+            if response.status_code == 200:
+                logger.info(str(len(bytearray(to_send_data_json))) + " bytes of data are reported.")
+                #logger.info(str(len(bytearray(to_send_data_json))) + " ------------------------------------------------------------------")
+
+                logger.debug("--- Send data time: %s seconds ---" % (time.time() - send_data_time))
+            else:
+                logger.info("Failed to send data.")
+            return
+        except requests.exceptions.Timeout:
+            logger.exception(
+                "Timed out while flushing to InsightFinder. Reattempting...")
+            continue
+        except requests.exceptions.TooManyRedirects:
+            logger.exception(
+                "Too many redirects while flushing to InsightFinder.")
+            break
+        except requests.exceptions.RequestException as e:
+            logger.exception(
+                "Exception while flushing to InsightFinder.")
+            break
+
+    logger.error(
+        "Failed to flush to InsightFinder! Gave up after %d attempts.", ATTEMPTS)
```

Comment:
replace _ in the for statement and use that iterator variable instead of ATTEMPTS

Reply:
The iterator variable is used only in the format_data() function to off set the dictionary to avoid the first two metrics namely container_id and host.HostName. While the Attempts variable is used to handle retrys. 
The _  was used  in the for loop as the value of ATTEMPTS variable was not used anywhere else.

Reply:
I wasn't talking about that loop. I'm talking about the case where the loop breaks early - requests.exceptions.TooManyRedirects or requests.exceptions.RequestException. It could have given up after 1 or 2 tries.